jolt-core/src/jolt/vm/read_write_memory.rs

```diff
@@ -71,17 +75,54 @@ where
 
 impl<F: PrimeField, G: CurveGroup<ScalarField = F>> ReadWriteMemory<F, G> {
     pub fn new(
+        bytecode: Vec<ELFInstruction>,
         memory_trace: Vec<MemoryOp>,
-        memory_size: usize,
         transcript: &mut Transcript,
     ) -> (Self, Vec<u64>) {
         let m = memory_trace.len();
         assert!(m.is_power_of_two());
 
+        let remap_address = |a: u64| {
+            assert!(a < REGISTER_COUNT || a >= RAM_START_ADDRESS);
+            if a >= RAM_START_ADDRESS {
+                a - RAM_START_ADDRESS + REGISTER_COUNT
+            } else {
+                // If a < REGISTER_COUNT, it is one of the registers and doesn't
+                // need to be remapped
+                a
+            }
+        };
+
+        let max_memory_address = memory_trace
+            .iter()
+            .map(|op| match op {
+                MemoryOp::Read(a, _) => remap_address(*a),
+                MemoryOp::Write(a, _) => remap_address(*a),
+            })
+            .max()
+            .unwrap_or(0);
+        let max_bytecode_address = bytecode
+            .iter()
+            .map(|instr| remap_address(instr.address + 3))
```

Comment:
Could we move the + 3 to after the max and add a comment along the lines: "For RV32, 4 bytes of memory addressable raw instructions, max address of bytecode would be instruction + 3."